game-core/src/main/java/games/strategy/engine/data/GameParser.java

```diff
@@ -1254,48 +1253,25 @@ private Attachable findAttachment(final Element element, final String type) thro
     return returnVal;
   }
 
-  private static String capitalizeFirstLetter(final String input) {
-    char first = input.charAt(0);
-    first = Character.toUpperCase(first);
-    return first + input.substring(1);
-  }
-
   private ArrayList<Tuple<String, String>> setValues(final IAttachment attachment, final List<Element> values)
       throws GameParseException {
+    final Map<String, Function<IAttachment, AttachmentProperty<?>>> attachmentMap = attachment.getPropertyMap();
     final ArrayList<Tuple<String, String>> options = new ArrayList<>();
     for (final Element current : values) {
       // find the setter
       final String name = current.getAttribute("name");
       if (name.length() == 0) {
         throw newGameParseException("Option name with 0 length");
       }
-      final Method setter;
-      try {
-        setter = attachment.getClass().getMethod("set" + capitalizeFirstLetter(name), SETTER_ARGS);
-      } catch (final NoSuchMethodException nsme) {
-        throw newGameParseException(String.format(
-            "The following option name of %s of class %s are either misspelled"
-                + " or exist only in a future version of TripleA. Setter: %s",
-            attachment.getName(), attachment.getClass().getSimpleName(), name));
-      }
       // find the value
       final String value = current.getAttribute("value");
       final String count = current.getAttribute("count");
-      final String itemValues;
-      if (count.length() > 0) {
-        itemValues = count + ":" + value;
-      } else {
-        itemValues = value;
-      }
-      // invoke
-      try {
-        setter.invoke(attachment, itemValues);
-      } catch (final IllegalAccessException iae) {
-        throw newGameParseException(
-            "Setter not public. Setter:" + name + " Class:" + attachment.getClass().getName(), iae);
-      } catch (final InvocationTargetException ite) {
-        throw newGameParseException("Error setting property:" + name, ite);
-      }
+      final String itemValues = (count.length() > 0 ? count + ":" : "") + value;
+      Optional.ofNullable(attachmentMap.get(name))
+          .orElseThrow(() -> new GameParseException("Missing property definition for option: " + name))
+          .apply(attachment)
+          .setValue(itemValues);
```

Comment:
Looking at this further... 

There is a feature envy code smell with `attachment`, we access data from it then operate on it to then send it back to attachment. This means we can move functionality to that class:  7a3d4ee

That would allow you to test this hunk of code:
```
  default void setValues(final List<Element> values)
      throws GameParseException {
    final Map<String, Function<IAttachment, AttachmentProperty<?>>> attachmentMap = getPropertyMap();
    for (final Element current : values) {
      // find the setter
      final String name = current.getAttribute("name");
      if (name.length() == 0) {
        throw new GameParseException("Option name with 0 length");
      }
      // find the value
      final String value = current.getAttribute("value");
      final String count = current.getAttribute("count");
      final String itemValues = (count.length() > 0 ? count + ":" : "") + value;
      Optional.ofNullable(attachmentMap.get(name))
          .orElseThrow(() -> new GameParseException("Missing property definition for option: " + name))
          .apply(this)
          .setValue(itemValues);
    }
  }
```
It would then be possible to test, in a test context one would construct a new instance of `IAttachment` as an anonymous instance, the implementation is then the data stub which can include mocks, then call `setValues` and verify you get expected results. It would be nicer for `setValues` to not be a void method. It would be worth looking at if it made sense splitting it up into two stages, one where data is extracted and a second where it is set.

Then, I would recommend to inline the property map to make it clear it i is configuration, I'll comment on that in-context.